report/sections/TestRunner/chapter.tex

```diff
@@ -1,33 +1,32 @@
 \chapter{Test Runner} \label{chap:TestRunner}
-In chapter \ref{chap:TestRunner}, we described how running multiple GHC interpreter processes asynchronously eventually crashes the Test Runner web server given sufficiently many simultaneous requests.
-As mentioned, we needed a way to limit the number of running GHC interpreter processes in order to prevent crashes.
-
-The Test Runner exposes an endpoint \texttt{/haskell/submit/}, which is used to post code submissions.
-Originally, we used the Rust crate Rocket to implement this endpoint.
-However, we eventually switched to using the axum crate instead, as it is developed and maintained by the same developers behind the Tokio package we make heavy use of for asynchronous programming and multithreading.
-Furthermore, it was deemed easier to implement a shared state among endpoints by leveraging axum's \texttt{layer} feature.
-We give an overview and a more technical explanation of how this was done in section \ref{sec:queue-system}.
-
-When the Test Runner receives the request from Next.js, its singular goal is to run the tests against the code and send the result of this back as fast as possible.
-In order to run the tests, we need to run the GHC interpreter on the test code, which, in turns, need access to the exercise submission code.
+When the Test Runner receives the request from the backend component, its singular goal is to run the tests against the code and send back the result as fast as possible.
+In order to run the tests, we run the GHC interpreter on the test code, which, in turns, need access to the exercise submission code.
 The GHC interpreter can be run from the command line with arguments specifying the path Haskell file to run.
-In other words, the Test Runner needs to write the user submission and corresponding test code to a file.
-However, if more users were to make a submission at the same time, the content of the file could potentially be overwritten while the GHC interpreter is running the tests.
-
+In other words, the Test Runner writes the user submission and corresponding test code to a file.
```

Comment:
```suggestion
So to run the tests, the Test Runner writes the user's code and corresponding exercise test code to a file.
```
This feels clearer. Removing `needs` from the sentence changed its meaning, which IMO made it a bit wonky.

And should we clarify that it's actually writing to separate files?

Reply:
```suggestion
So to run the tests, the Test Runner writes the code submitted by the user, and the accompanying exercise test code, to a file.
```

Reply:
```suggestion
So to run the tests, the Test Runner writes the code submitted by the user, and the accompanying exercise test code, to files.
```
I don't think `a file each` is great. What do you think of this formulation?
or `separate files`?

Reply:
```suggestion
So to run the tests, the Test Runner writes the code submitted by the user, and the accompanying exercise test code, to separate files.
```